### PR TITLE
Functional tests for #302

### PR DESCRIPTION
**Pull request by rynowak**
This feature allows routing to generate a link when the action that is
being linked-to does not yet exist. See the PR in routing for the actual
implementation changes. This PR just has a simple functional test for the
scenario.

### CODE REVIEW

**Comment by rynowak** [comment]
See https://github.com/aspnet/Routing/pull/150

**Comment by rynowak** [comment]
@harshgMSFT  ping

File: test/Microsoft.AspNet.Mvc.FunctionalTests/BestEffortLinkGenerationTest.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            var url = "http://localhost/Home/Index";
+            // Act
+            var response = await client.GetAsync(url);
+            var content = await response.Content.ReadAsStringAsync();
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal(ExpectedOutput, content);
+        }
+
```
Comment (harshgMSFT) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/BestEffortLinkGenerationTest.cs]: nit remove extra line.

File: test/WebSites/BestEffortLinkGenerationWebSite/project.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        "Microsoft.AspNet.Mvc.TestConfiguration": "1.0.0",
+        "Microsoft.AspNet.Server.IIS": "1.0.0-*",
+        "Microsoft.AspNet.Server.WebListener": "1.0.0-*",
+        "Microsoft.AspNet.StaticFiles": "1.0.0-*"
+    },
+    "frameworks": {
+        "aspnet50": { },
+        "aspnetcore50": { }
+    },
+    "webroot": "wwwroot"
```
Comment (harshgMSFT) [inline test/WebSites/BestEffortLinkGenerationWebSite/project.json]: add readme.md

**Comment by harshgMSFT** [comment]
:shipit:

**Comment by rynowak** [comment]
thanks. The routing change for this is in. Currently waiting on a coherence pass to check in this test

File: test/Microsoft.AspNet.Mvc.FunctionalTests/BestEffortLinkGenerationTest.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/BestEffortLinkGenerationTest.cs]: Extra line

File: test/WebSites/BestEffortLinkGenerationWebSite/Startup.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                services.AddMvc(configuration);
+                services.Configure<RouteOptions>((options) =>
+                {
+                    options.UseBestEffortLinkGeneration = true;
+                });
+            });
+
+            app.UseMvc(routes =>
+            {
+                routes.MapRoute(
```
Comment (pranavkm) [inline test/WebSites/BestEffortLinkGenerationWebSite/Startup.cs]: Attribute route?

Comment (rynowak) [inline test/WebSites/BestEffortLinkGenerationWebSite/Startup.cs]: This is a feature specifically related to conventional routes

**Pull request merged**